### ISSUE TITLE
parent is not defined error when creating a relation in node SDK

### ISSUE THREAD

**Issue by estock** [ParsePlatform/Parse-SDK-JS#4]
I am executing the following code 

```
  var TestObject = Parse.Object.extend("TestObject");
  var testObject = new TestObject();
  testObject.set("Name", "testName");

  testObject.save(null, {
    success: function(testObject) {
      // Execute any logic that should take place after the object is saved.
      console.log('New testObject created with objectId: ' + testObject.id);
      var TestObject2 = Parse.Object.extend("TestObject2");
      var testObject2 = new TestObject2();
      testObject2.set("Name", "testName2");
      testObject2.save(null, {
        success: function(testObject2) {
          // Execute any logic that should take place after the object is saved.
          console.log('New testObject2 created with objectId: ' + testObject2.id);
          var relation = testObject.relation("TestObjectRelation");
          console.log('Make sure the relation exists ' + testObject.relation("TestObjectRelation"));

          relation.add(testObject2);
          testObject.save();
        },
        error: function(testObject2, error) {
          // Execute any logic that should take place if the save fails.
          // error is a Parse.Error with an error code and message.
          console.log('Failed to create new testObject2, with error code: ' + error.message);
        }
      });

    },
    error: function(venue, error) {
      // Execute any logic that should take place if the save fails.
      // error is a Parse.Error with an error code and message.
      console.log('Failed to create new testObject, with error code: ' + error.message);
    }
  });
```

I create an object, and then create a second object and try to save that as a relation to the first object ... though I am following the documented syntax, I always get the same error 

ReferenceError: parent is not defined
    at ParseRelation.add (/home/showcast/node_modules/parse/lib/node/ParseRelation.js:102:14)

Is this a bug in the node SDK ? are parse it's self?

**Comment by andrewimm** [comment]
We've had reports of this bug in the past, but never had a reproducible test case. I'm currently looking into it.

**Comment by stanleyw** [comment]
Previously reported on FB bug tool: https://developers.facebook.com/bugs/1677477152487599/

**Comment by neolution** [comment]
I can confirm this issue.

Here is my workaround for "cloud code":
change the "jssdk"-parameter to "1.5.0" in the file "parse.project"-file.

{
  "parse": {
    "jssdk": "1.5.0"
  },
}

Redeploy with "parse deploy" and check again.

This could't only be a temporary fix.

**Comment by andrewimm** [comment]
Resolved in 1.6.4
